flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobManagerSharedServices.java

```diff
@@ -147,11 +157,23 @@ public static JobManagerSharedServices fromConfiguration(
                                 failOnJvmMetaspaceOomError ? fatalErrorHandler : null,
                                 checkClassLoaderLeak));
 
+        final int numJobManagerFutureThreads =
+                config.getInteger(RestOptions.JOBMANAGER_FUTURE_THREADS);
```

Comment:
If you remove the default value from this option, then you can do something like this:

`config.getInteger(RestOptions.JOBMANAGER_FUTURE_THREADS, Hardware.getNumberCPUCores())`

Reply:
Removing default value from config to remove this logic.